src/natcap/invest/seasonal_water_yield/seasonal_water_yield.py

```diff
@@ -1081,7 +1085,7 @@ def cn_op(lulc_array, soil_group_array):
             raise ValueError(error_message)
 
         for soil_group_id in numpy.unique(soil_group_array):
-            if soil_group_id == soil_nodata:
+            if utils.array_equals_nodata(numpy.array(soil_group_id), soil_nodata).any():
                 continue
             current_soil_mask = (soil_group_array == soil_group_id)
```

Comment:
Sorry to barge in, but what about a second function `utils.number_equals_nodata` to more neatly check equality of a single number? Or replace both with a function `utils.equals_nodata`, which does the appropriate comparison based on whether the input is an array or a number?

Reply:
Yeah I think that's a good idea. But also then it might follow to have a generic `utils.equals_number` because sometimes (like in this `soil_group_array == soil_group_id`) we're not technically dealing with nodata. And then it really starts to feel like we're re-inventing the wheel and obfuscating things by not just using the APIs that python/numpy gives us.

Reply:
Yea, I didn't love doing this either.
> @davemfish theoretically should we be guarding against floats and `nan` here too? 

Great question and one I came across often. There were a lot of instances of model constant defined nodata values where I knew it couldn't be `numpy.nan`. But there were a lot of these instances where I couldn't be sure unless I did some investigating and type checking, which for me, conflicted against this being a function to help you avoid that thought process.

> @emlys Or replace both with a function `utils.equals_nodata`

I thought about a separate function but that somehow felt like a moment where we should think about this a bit more. Having one function seems better, where the input can be either number or `numpy.array`.